src/components/Carousel.tsx

```diff
@@ -30,46 +38,68 @@ function Carousel(props: CarouselProps): JSX.Element {
     storage.setItem('slideIdx', slideIdx.toString());
   }, [slideIdx]);
 
+  function goNext(): void {
+    setSlideIdx(old => Math.min(old + 1, props.children.length - 1));
+    props.onNext && props.onNext();
+  }
+
+  function goPrev(): void {
+    setSlideIdx(old => Math.max(old - 1, 0));
+    props.onPrev && props.onPrev();
+  }
+
+  const child = props.children[slideIdx];
+  if (child === undefined && props.children.length > 0) {
+    const slideInBounds = Math.min(Math.max(0, slideIdx), props.children.length - 1);
+    setSlideIdx(slideInBounds);
+  }
+
   return (
-    <div id={'carousel-wrapper'}>
-      { props.title && <h1 id={'title'}>{props.title}</h1> }
-      { props.subtitle && <h2 id={'subtitle'}>{props.subtitle}</h2> }
-      <div id={'carousel'}>
-        <button
-          className={'carousel-btn prev'}
-          style={{
-            visibility: ((props.showPrev !== undefined && props.showPrev) || (slideIdx > 0))
-              ? 'visible'
-              : 'hidden',
-          }}
-          onClick={() => {
-            setSlideIdx(old => Math.max(old - 1, 0));
-            props.onPrev && props.onPrev();
-          }}
-        >
-          <img src={PrevSvg} />
-        </button>
-        <div id={'carousel-content'}>
-          {props.children && props.children.length > 0 && props.children[slideIdx]}
-        </div>
-        <button
-          className={'carousel-btn next'}
-          style = {{
-            visibility:
-              ((props.showNext !== undefined && props.showNext) ||
-               (slideIdx < (props.children ? props.children.length - 1 : 0)))
+    <ChangeSlideContext.Provider value={{next: goNext, prev: goPrev}}>
+      <div id={'carousel-wrapper'}>
+        {props.title && <h1 id={'title'}>{props.title}</h1>}
+        {props.subtitle && <h2 id={'subtitle'}>{props.subtitle}</h2>}
+        <div id={'carousel'}>
+          <button
+            className={'carousel-btn prev'}
+            style={{
+              visibility: child &&
+                ((child.showPrev === undefined && slideIdx > 0) ||
+                  (child.showPrev === true))
```

Comment:
thanks! this is a lot more readable than the last one!

nit:
```suggestion
              visibility: child?.showPrev === false || slideIdx === 0 ? 'hidden' : 'visible'
```

the `?` operator is syntactic sugar to see if the child is defined before resolving the property

Reply:
genius :0